### PR TITLE
Add config and logger

### PR DESCRIPTION
**Pull request by maxh65000**

### CODE REVIEW

File: deployconfig.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+scp src/main/deploy/config.properties admin@10.1.38.28:/home/lvuser/deploy/config.properties
```
Comment (Team138Entropy) [inline deployconfig.sh]: Assuming this should be 10.1.38.28

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -1,38 +1,35 @@
-package frc.robot;
+// package frc.robot;
 /*
  * Constant values used throughout robot code.
  * In "C" this would be a header file, but alas, this is Java
  */
-public class Constants {
+// @Deprecated
+// public class Constants {
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Constants.java]: Let's reinstate the Constants class and create enumerations for each of the general constants categories. 
Use the enum values to index a dictionary that holds the string key for the constant.
You can then encapsulate the config store inside the Constants class, thereby abstracting the whole mechanism for retrieving constants.

File: src/main/java/frc/robot/commands/TeleopDrive.java [hunk 3]
```diff
@@ -15,6 +17,7 @@ public TeleopDrive(){
 	}
 
 	protected void initialize() {
+		teleopLogger.silly("");
```
Comment (jhinspeter) [inline src/main/java/frc/robot/commands/TeleopDrive.java]: What is this? Should there be a message?

File: src/main/java/frc/robot/Config.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+
+    /**
+     * 
+     * @param key The name of the property. It should be camelCase and use periods for depth, eg. user.name
+     * @return The value of the property casted to a float.
+     */
+    public float getFloat(String key){
+        return Float.parseFloat(getProp(key));
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Config.java]: There's no exception handling here! 
What if the key doesn't exist? What if parseFloat fails?

**Review by jhinspeter — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Constants.java (hunk 2 above)
Comment (maxh65000) [reply src/main/java/frc/robot/Constants.java]: I made the enum thing work.

File: src/main/java/frc/robot/Config.java (hunk 4 above)
Comment (maxh65000) [reply src/main/java/frc/robot/Config.java]: It throws smh

File: src/main/java/frc/robot/commands/TeleopDrive.java (hunk 3 above)
Comment (maxh65000) [reply src/main/java/frc/robot/commands/TeleopDrive.java]: Addressed

File: src/main/java/frc/robot/Config.java (hunk 4 above)
Comment (maxh65000) [reply src/main/java/frc/robot/Config.java]: We want the robot to crash so that we can fix our problems easier

**Review by Team138Entropy — APPROVED** [review]

**Review by jhinspeter — APPROVED** [review]

File: src/main/java/frc/robot/Logger.java [hunk 5]
```diff
@@ -53,16 +55,18 @@ public static String getLogMsg(String logPath, String thisLogLevel, String messa
...
 
         // if logging is enabled at all for this logger,
         // and the level is recognized,
         // and our level is higher-up or equal to the minimum specified,
         // then log it
         if(minLevel != null && minLevel != "OFF" && levels.contains(level) && levels.indexOf(level) <= levels.indexOf(minLevel)){
             // if the level we're logging at is WARN or ERR, then log to STDERR, otherwise log to STDOUT
-            if(level == "WARN" || level == "ERR"){
-                System.err.println(getLogMsg(logPath, level, message));
+            if(level == "ERROR"){
```
Comment (Team138Entropy) [inline src/main/java/frc/robot/Logger.java]: Level should be an enumeration so problems are caught at compile time rather than runtime.

**Pull request merged**